pkg/config/validation/validation.go

```diff
@@ -798,6 +798,12 @@ var ValidateEnvoyFilter = registerValidateFunc("ValidateEnvoyFilter",
 			return nil, err
 		}
 
+		// If workloadSelector is defined and labels are not set, it is most likely
+		// an user error. Marking it as a warning to keep it backwards compatible.
+		if rule.WorkloadSelector != nil && len(rule.WorkloadSelector.Labels) == 0 {
+			errs = appendValidation(errs, WrapWarning(fmt.Errorf("Envoy filter: labels not defined in workload selector"))) // nolint: stylecheck
```

Comment:
nit: change the message as 
"Envoy filter:  workload selector specified with out labels. This will filter will be applied to services in the namespace"